econml/iv/dml/_dml.py

```diff
@@ -495,9 +897,55 @@ def __init__(self, discrete_instrument=False, discrete_treatment=False, categori
                          mc_agg=mc_agg,
                          random_state=random_state)
 
-    @_deprecate_positional("Z and X should be passed by keyword only. In a future release "
-                           "we will disallow passing Z and X by position.", ['X', 'Z'])
-    def fit(self, Y, T, Z, X=None, *, sample_weight=None, freq_weight=None, sample_var=None, groups=None,
+    def _gen_featurizer(self):
+        return clone(self.featurizer, safe=False)
+
+    def _gen_model_y_xw(self):
+        if self.model_y_xw == 'auto':
+            model_y_xw = WeightedLassoCVWrapper(random_state=self.random_state)
+        else:
+            model_y_xw = clone(self.model_y_xw, safe=False)
+        return _FirstStageWrapper(model_y_xw, True, self._gen_featurizer(),
+                                  False, False)
+
+    def _gen_model_t_xw(self):
+        if self.model_t_xw == 'auto':
+            if self.discrete_treatment:
+                model_t_xw = LogisticRegressionCV(cv=WeightedStratifiedKFold(random_state=self.random_state),
+                                                  random_state=self.random_state)
+            else:
+                model_t_xw = WeightedLassoCVWrapper(random_state=self.random_state)
+        else:
+            model_t_xw = clone(self.model_t_xw, safe=False)
+        return _FirstStageWrapper(model_t_xw, False, self._gen_featurizer(),
+                                  False, self.discrete_treatment)
+
+    def _gen_model_t_xwz(self):
+        if self.model_t_xwz == 'auto':
+            if self.discrete_treatment:
+                model_t_xwz = LogisticRegressionCV(cv=WeightedStratifiedKFold(random_state=self.random_state),
+                                                   random_state=self.random_state)
+            else:
+                model_t_xwz = WeightedLassoCVWrapper(random_state=self.random_state)
+        else:
+            model_t_xwz = clone(self.model_t_xwz, safe=False)
+        return _FirstStageWrapper(model_t_xwz, False, self._gen_featurizer(),
+                                  False, self.discrete_treatment)
+
+    def _gen_model_final(self):
+        return clone(self.model_final, safe=False)
+
+    def _gen_ortho_learner_model_nuisance(self):
+        return _BaseDMLIVModelNuisance(self._gen_model_y_xw(), self._gen_model_t_xw(), self._gen_model_t_xwz())
+
+    def _gen_ortho_learner_model_final(self):
+        return _BaseDMLIVModelFinal(_FinalWrapper(self._gen_model_final(),
+                                                  self.fit_cate_intercept,
+                                                  self._gen_featurizer(),
+                                                  False))
+
+    # override only so that we can enforce Z to be required
+    def fit(self, Y, T, *, Z, X=None, W=None, sample_weight=None, freq_weight=None, sample_var=None, groups=None,
```

Comment:
It seems that this fit, could just go up to the _BaseDMLIV and not be repeated here and in the NonParamDMLIV

Reply:
you are right! Done